test/scenarios/ynEIGEN/ynLSDeUpgradeScenario.sol

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: BSD 3-Clause License
+pragma solidity ^0.8.24;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ITransparentUpgradeableProxy} from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {UpgradeableBeacon} from "@openzeppelin/contracts/proxy/beacon/UpgradeableBeacon.sol";
+
+import {ITokenStakingNode} from "../../../src/interfaces/ITokenStakingNode.sol";
+
+import {HoleskyLSDRateProvider} from "../../../src/testnet/HoleksyLSDRateProvider.sol";
+
+import {TestStakingNodesManagerV2} from "../../mocks/TestStakingNodesManagerV2.sol";
+import {TestStakingNodeV2} from "../../mocks/TestStakingNodeV2.sol";
+
+import "./ynLSDeScenarioBaseTest.sol";
+import "forge-std/console.sol";
+
+contract ynLSDeUpgradeScenario is ynLSDeScenarioBaseTest {
+
+    address public constant TIMELOCK = 0x62173555C27C67644C5634e114e42A63A59CD7A5;
+    
+    function test_Upgrade_ynLSDe_Scenario() public {
+        if (block.chainid != 17000) return;
+
+        address previousImplementation = getTransparentUpgradeableProxyImplementationAddress(address(yneigen));
+        address newImplementation = address(new ynEigen()); 
+
+        uint256 previousTotalAssets = yneigen.totalAssets();
+        uint256 previousTotalSupply = IERC20(address(yneigen)).totalSupply();
+
+        vm.prank(getTransparentUpgradeableProxyAdminAddress(address(yneigen)));
```

Comment:
The pranking here goes too far, since if there's any issues with the ProxyAdmin or the timelockcontroller this test still passes.

The mocking should be only at the level of the gnosis multisigs which are assumed to always work correctly.



For this test to be useful we should do what you did here:

https://github.com/yieldnest/yieldnest-protocol/pull/116/files#diff-e0a04b3d30c78959e47965a5312ec325ce53d171f2cc67e9191055bd5245fce0R50

To ensure the whole pipeline works correctly, along with encoding the tx, advancing time etc.

Feel free to define the utilities you need

Reply:
to keep in mind the delay is different on holesky vs mainnet

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/150/commits/4d2848b4d96223b196d40303ad13dc4db9516ce7